core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -708,7 +711,8 @@ private void failReplicaIfNeeded(String index, int shardId, Throwable t) {
          */
         public ReplicationPhase(ShardIterator originalShardIt, ReplicaRequest replicaRequest, Response finalResponse,
                                 ClusterStateObserver observer, ShardRouting originalPrimaryShard,
-                                InternalRequest internalRequest, ActionListener<Response> listener, Releasable indexShardReference) {
+                                InternalRequest internalRequest, ActionListener<Response> listener, Releasable indexShardReference,
+                                ShardStateActionListener shardStateListener) {
```

Comment:
I'm wondering if the ReplicaPhase shouldn't implement the ShardStateActionListener interface it self, which will make things simpler and save on the param.